solr/core/src/java/org/apache/solr/cluster/placement/ShardMetrics.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cluster.placement;
+
+import java.util.Optional;
+
+/**
+ *
+ */
+public interface ShardMetrics {
+  Optional<ReplicaMetrics> getLeaderMetrics();
+  Optional<ReplicaMetrics> getReplicaMetrics(String replicaName);
```

Comment:
Perhaps we should add `iterator()` here too, so that the consumers are not required to know the replica name in advance.

Reply:
Yes. As well as `ShardMetrics` iterator in `CollectionMetrics`.